src/main/java/util/miscellaneous/DukeOperation.java

```diff
@@ -0,0 +1,321 @@
+            c = CommandType.SAVE;
+        } else {
+            c = CommandType.NIL;
+        }
+
+        return c;
+    }
+
+    public static void checkCommand(ArrayList<Task> tasks,String line, CommandType c) throws NoTaskException, NoDateException, NoItemException {
+        switch (c) {
+        case TODO:
+            String todo = line.substring(TODO_TASK_INDEX);
+
+            if ((todo.trim()).isEmpty()) {
+                throw new NoTaskException();
+            }
+
+            break;
+        case DEADLINE:
+            String by = line.substring(line.indexOf(DEADLINE_OF_TASK_CMD) + TIME_INDEX);
+            String deadline = line.substring(DEADLINE_TASK_INDEX, line.indexOf(DEADLINE_OF_TASK_CMD));
+
+            if ((deadline.trim()).isEmpty()) {
+                throw new NoTaskException();
+            }
+
+            if ((by.trim()).isEmpty()) {
+                throw new NoDateException();
+            }
+            break;
+        case EVENT:
+            String at = line.substring(line.indexOf(DURATION_OF_EVENT_CMD) + TIME_INDEX);
+            String event = line.substring(EVENT_TASK_INDEX, line.indexOf(DURATION_OF_EVENT_CMD));
+
+            if ((event.trim()).isEmpty()) {
+                throw new NoTaskException();
+            }
+
+            if ((at.trim()).isEmpty()) {
+                throw new NoDateException();
+            }
+
+            break;
+        case MARK:
+            int markedItem = Integer.parseInt(line.substring(MARKED_ITEM_INDEX)) - 1;
+
+            if ((markedItem < 0) || (markedItem >= tasks.size())) {
+                throw new NoItemException();
+            }
+
+            break;
+        case UNMARK:
+            int unmarkedItem = Integer.parseInt(line.substring(UNMARKED_ITEM_INDEX)) - 1;
+
+            if ((unmarkedItem < 0) || (unmarkedItem >= tasks.size())) {
+                throw new NoItemException();
+            }
+
+            break;
+        case DEL:
+            int index = Integer.parseInt(line.substring(DELETE_INDEX)) - 1;
+
+            if ((index < 0) || (index >= tasks.size())) {
+                throw new NoItemException();
+            }
+
+            break;
+        default:
+            break;
+        }
+    }
+
+    public static int handleError(ArrayList<Task> tasks,String line, CommandType c) {
+        try {
+            checkCommand(tasks, line, c);
+        } catch (IndexOutOfBoundsException e01) {
+            echo(ITEM_NOT_EXIST_MSG);
+            return 1;
+        } catch (NoDateException e02) {
+            echo(NO_DATE_MSG);
+            return 1;
+        } catch (NoTaskException e03) {
+            echo(NO_TASK_MSG);
+            return 1;
+        } catch (NoItemException e04) {
+            echo(ITEM_NOT_EXIST_MSG);
+            return 1;
+        } finally {
+            return 0;
+        }
+    }
+
+    public static void loadAndRun(ArrayList<Task> tasks, String line, boolean isLoadingData, boolean needUpdateTaskStatus) {
```

Comment:
This method is too long ([>30 LoC](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)). How about apply [SLAP](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) here to increase code readability?